sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -310,7 +310,14 @@ class Dataset[T] private[sql](
           case binary: Array[Byte] => binary.map("%02X".format(_)).mkString("[", " ", "]")
           case _ =>
             // Escapes meta-characters not to break the `showString` format
-            cell.toString.replaceAll("\n", "\\\\n").replaceAll("\t", "\\\\t")
+            cell.toString
+              .replaceAll("\n", "\\\\n")
+              .replaceAll("\r", "\\\\r")
+              .replaceAll("\t", "\\\\t")
+              .replaceAll("\f", "\\\\f")
+              .replaceAll("\b", "\\\\b")
+              .replaceAll("\u000B", "\\\\v")
+              .replaceAll("\u0007", "\\\\a")
```

Comment:
It seems `fieldNames` (L306) has the same issue, so could you fix it in this PR, too?
```
scala> spark.range(1).selectExpr("'aaa\nbbb'").show()
+--------+                                                                      
| aaa
bbb|
+--------+
|aaa\nbbb|
+--------+
```

Reply:
Sure, will do.